### PR TITLE
added things from bose

### PR DESCRIPTION
**Pull request by paulrolfe**
- Vertical gradient layer
- to array from array slice
- some geometry helpers like `ip_topLeft`

### CODE REVIEW

File: SwiftWisdom/Core/Geometry/Geometry.swift [hunk 1]
```diff
@@ -196,3 +196,26 @@ extension CGSize {
         return fixedHeight * scale
     }
 }
+
+extension CGRect {
```
Comment (loganwright) [inline SwiftWisdom/Core/Geometry/Geometry.swift:200]: Yes!

File: SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift [hunk 2]
```diff
@@ -35,6 +35,12 @@ public extension Array where Element: Hashable {
     }
 }
 
+extension ArraySlice {
+    public func ip_toArray -> Array<Element> {
```
Comment (loganwright) [inline SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift]: does this compile?

File: SwiftWisdom/Core/UIKit/CoreAnimation/BasicVerticalGradientLayer.swift [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+//  CAGradientLayer.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 2/17/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import UIKit
+
+final class BasicVerticalGradientLayer : CAGradientLayer {
```
Comment (loganwright) [inline SwiftWisdom/Core/UIKit/CoreAnimation/BasicVerticalGradientLayer.swift]: `public final`

File: SwiftWisdom/Core/UIKit/CoreAnimation/BasicVerticalGradientLayer.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    init(topColor: UIColor = .grayColor(), bottomColor: UIColor = .blackColor()) {
```
Comment (loganwright) [inline SwiftWisdom/Core/UIKit/CoreAnimation/BasicVerticalGradientLayer.swift]: `public`

**Comment by loganwright** [comment]
Comments ... Bump ... Ship It

File: SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift (hunk 2 above)
Comment (paulrolfe) [inline SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift]: my b.

**Pull request merged**